syntax = "proto3";

package ohmydb;

service OhMyDB {
    rpc TestCall(Cmd) returns(Ack) {}
    rpc Put(PutRequest) returns(PutResponse) {}
    rpc Get(GetRequest) returns(GetResponse) {}
}

message Ack {
    int32 ok = 1;
}

message Cmd {
    int32 sup = 1;
}

message PutRequest{
    int32 key = 1;
    int32 value = 2;
}

message PutResponse {
    int32 error_code = 1;
    string leader_addr = 2;
}

message GetRequest{
    int32 key = 1;
}

message GetResponse{
    int32 error_code = 1;
    string leader_addr = 2;
    int32 value = 3;
}